src/build_workflow/build_recorder.py

```diff
@@ -53,18 +53,25 @@ def write_manifest(self) -> None:
         logging.info(f"Created build manifest {manifest_path}")
 
     class BuildManifestBuilder:
-        def __init__(self, target: BuildTarget) -> None:
+        def __init__(self, target: BuildTarget, build_manifest: BuildManifest = None) -> None:
             self.data: Dict[str, Any] = {}
-            self.data["build"] = {}
-            self.data["build"]["id"] = target.build_id
-            self.data["build"]["name"] = target.name
-            self.data["build"]["version"] = target.opensearch_version
-            self.data["build"]["platform"] = target.platform
-            self.data["build"]["architecture"] = target.architecture
-            self.data["build"]["distribution"] = target.distribution if target.distribution else "tar"
-            self.data["schema-version"] = "1.2"
             self.components_hash: Dict[str, Dict[str, Any]] = {}
 
+            if build_manifest:
+                build_manifest_data = build_manifest.__to_dict__()
+                self.data = build_manifest_data
```

Comment:
I believe you should be able to iterate through components without turning the build manifest into a dictionary first, which would remove the need for the `build_manifest_data` variable.

Reply:
I think that would require to change the logics in other sections for example `to_manifest`  from `build_recorder` since it generates the build manifest based on dictionary. Do you think it's necessary or it may be overcomplicated.

Reply:
Whatever you need to add to `build_manifest` is a good idea. It should look like this here:

```python
self.data = dict(build_manifest)
for component in build_manifest.components:
    self.components_hash[component.name] = dict(component)
```

Reply:
I see what you mean! Updated in the latest commit. Thanks for suggestion!